contracts/governance/extensions/GovernorNoncesKeyed.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Governor} from "../Governor.sol";
+import {Nonces} from "../../utils/Nonces.sol";
+import {NoncesKeyed} from "../../utils/NoncesKeyed.sol";
+import {SignatureChecker} from "../../utils/cryptography/SignatureChecker.sol";
+
+/**
+ * @dev An extension of {Governor} that extends existing nonce management to use {NoncesKeyed}, where the key is the first 192 bits of the `proposalId`.
+ * This is useful for voting by signature while maintaining separate sequences of nonces for each proposal.
```

Comment:
Based on this definition, then it's possible (not sure if feasible) to "mine" a `key` by just adding a salt to the description of a proposal. I wonder if we should document further that a nonce can be invalidated if the user happens to sign a vote for a proposal with colliding first 192 bits.

Reply:
This is 4 bytes larger than an ethereum address--it would be significantly harder to mine a proposal nonce than choosing an arbitrary address (assuming fully random proposal id generation). I think we are fine here.

Reply:
Yeah lol, I underestimated how big this is. Solving